### PR TITLE
Add otel test options

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

NOTE: I meant to mark this as WIP.  Do not merge this yet.

### CODE REVIEW

**Comment by kevinearls** [comment]
@jpkrohling @pavolloffay @rubenvp8510  Please review

File: test/e2e/utils.go [hunk 1]
```diff
@@ -586,6 +656,26 @@ func getOtelCollectorOptions() map[string]interface{} {
 	return otelOptions
 }
 
+func getOtelAgentOptions() map[string]interface{} {
+	otelOptions := map[string]interface{}{
```
Comment (pavolloffay) [inline test/e2e/utils.go:660]: the map can be returned directly

File: test/e2e/utils.go [hunk 2]
```diff
@@ -586,6 +656,26 @@ func getOtelCollectorOptions() map[string]interface{} {
...
+		"extensions": map[string]interface{}{
+			"health_check": map[string]string{"port": "14269"},
+		},
+	}
+
+	return otelOptions
+}
+
+func getOtelAllInOneOptions() map[string]interface{} {
```
Comment (pavolloffay) [inline test/e2e/utils.go:669]: This func is the same as the `getOtelAgentOptions` I would unify and use just one `getOTELConfig(healthCheckPort int)`.

File: test/e2e/utils.go [hunk 3]
```diff
@@ -576,6 +582,70 @@ func wasUsingOtelCollector(jaegerInstanceName, namespace string) bool {
...
+		return false, nil
+	})
+	require.NoError(t, err)
+
+	if len(pods.Items) == 1 {
+		logrus.Infof("Found 1 pod %s", pods.Items[0].Name)
+		pod = pods.Items[0]
+	} else {
+		for _, p := range pods.Items {
+			if p.DeletionTimestamp == nil {
```
Comment (pavolloffay) [inline test/e2e/utils.go:628]: is there a way to query pods that are not scheduled for deletion (if I got correctly what `DeletionTimestamp` means)?

File: test/e2e/sidecar_test.go [hunk 4]
```diff
@@ -9,6 +9,8 @@ import (
 	"net/http"
 	"testing"
 
+	"github.com/sirupsen/logrus"
```
Comment (pavolloffay) [inline test/e2e/sidecar_test.go:12]: should be in the second group. Is the linter working?

**Review by pavolloffay — COMMENTED** [review]
Just some nits, LGTM

File: test/e2e/utils.go (hunk 3 above)
Comment (kevinearls) [reply test/e2e/utils.go:628]: You're correct, that's what we're looking for.  I don't know any way to do this.  @jpkrohling  do you?

File: test/e2e/sidecar_test.go (hunk 4 above)
Comment (kevinearls) [reply test/e2e/sidecar_test.go:12]: I don't think it works for tests

**Comment by kevinearls** [comment]
@pavolloffay  @jpkrohling If this looks ok now can you merge it?

File: test/e2e/utils.go (hunk 3 above)
Comment (jpkrohling) [reply test/e2e/utils.go:628]: I don't really know, perhaps there's something in the API docs about this?

File: test/e2e/utils.go [hunk 5]
```diff
@@ -576,14 +582,76 @@ func wasUsingOtelCollector(jaegerInstanceName, namespace string) bool {
...
+			logrus.Infof("Test %s is using agent image %s", t.Name(), container.Image)
+			return strings.Contains(container.Image, "jaeger-opentelemetry-agent")
+		}
+	}
+
+	require.Failf(t, "Did not find an agent image for %s in namespace %s", appName, namespace)
+	return false
+}
+
+func getOtelConfigOptions(healthCheckPort string) map[string]interface{} {
```
Comment (jpkrohling) [inline test/e2e/utils.go:649]: Isn't `healthCheckPort` being used? The function name is also confusing, perhaps `getOtelConfigForHealthCheckPort(healthCheckPort string)`? Or `getHealthCheckExtensionConfig()`?

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**